x-pack/plugin/wildcard/src/main/java/org/elasticsearch/xpack/wildcard/mapper/AutomatonQueryOnBinaryDv.java

```diff
@@ -99,12 +96,13 @@ public boolean equals(Object obj) {
             return false;
           }
         AutomatonQueryOnBinaryDv other = (AutomatonQueryOnBinaryDv) obj;
-        return Objects.equals(field, other.field)  && Objects.equals(matchPattern, other.matchPattern);
+        return Objects.equals(field, other.field)  && Objects.equals(matchPattern, other.matchPattern)
+            && Objects.equals(automaton, other.automaton);
```

Comment:
`Automaton` does not implement equals/hashCode. The `ByteRunAutomaton` should be used instead.